frontend/src/services/featureFlags/FeatureFlagManager.ts

```diff
@@ -125,25 +126,35 @@ export class FeatureFlagsManager {
   middleware(request: NextRequest, response: NextResponse): NextResponse {
     const paramValue = request.nextUrl.searchParams.get(FEATURE_FLAGS_KEY);
 
-    const featureFlagsFromQuery =
-      paramValue === "reset"
-        ? this.featureFlags
-        : parseFeatureFlagsFromString(paramValue);
+    if (paramValue === "reset") {
+      deleteCookie(response.cookies);
+      return response;
+    }
+
+    const featureFlagsFromQuery = parseFeatureFlagsFromString(paramValue);
 
     // previously there was logic here to return early if there were no feature flags active
     // beyond default values. Unfortunately, this breaks the implementation of the feature
-    // flag admin view, which depends on reading all flags from cookies, so the logic has beeen removed
-
-    const featureFlags =
-      environment.ENVIRONMENT === "prod"
-        ? { ...this.featureFlags }
-        : {
-            ...this.featureFlags,
-            ...getFeatureFlagsFromCookie(request.cookies),
-            ...featureFlagsFromQuery,
-          };
-
-    setCookie(JSON.stringify(featureFlags), response.cookies);
+    // flag admin view, which depends on reading all flags from cookies, so the logic has been removed
+
+    const featureFlags = {
```

Comment:
consider renaming to `userFeatureFlags` to more firmly differentiate from `this.featureFlags`?